File: frontend/src/api/playground.ts
import { apiFetch } from './client';

export interface PlaygroundRequest {
  config_id: string;
  message: string;
}

export interface PlaygroundResponse {
  text: string;
  latency_ms: number;
  token_usage: { input_tokens: number; output_tokens: number };
  raw_response: Record<string, unknown>;
}

export function runPlayground(data: PlaygroundRequest): Promise<PlaygroundResponse> {
  return apiFetch('/playground', { method: 'POST', body: JSON.stringify(data) });
}

File: src/ai_eval/routers/schemas/playground.py
"""Pydantic schemas for Playground endpoint."""

from pydantic import BaseModel


class PlaygroundRequest(BaseModel):
    """Request body for the playground endpoint."""

    config_id: str
    message: str


class PlaygroundResponse(BaseModel):
    """Full response from a playground invocation."""

    text: str
    latency_ms: int
    token_usage: dict
    raw_response: dict
